### PR TITLE
Port to GridTools 2.0

### PR DESCRIPTION
**Pull request by fthaler**
Port to GridTools 2.0. Further changes:
- Removed architecture-dependent block-size tuning, may be reintroduced as presets or tunable CMake-variables in the future.
- Updated CI to CUDA 11.0 and Ubuntu 20.04 LTS.
- Introduced HIP-builds in the CI setup.
- Renamed resulting Docker image tags to be consistent with those changes.
- Use Docker BuildKit for better caching support and thus faster CI.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+ARG BUILD_FROM=ubuntu:20.04
 FROM ${BUILD_FROM} as base
 LABEL maintainer="Felix Thaler <thaler@cscs.ch>"
 
 RUN apt-get update -qq && \
-    apt-get install -qq -y \
+    DEBIAN_FRONTEND=noninteractive apt-get install -qq -y \
     build-essential \
-    wget \
+    file \
```
Comment (fthaler) [inline Dockerfile:8]: `file` is required for hipcc but does not come preinstalled on the ROCm container.

File: Dockerfile [hunk 2]
```diff
@@ -24,32 +26,24 @@ RUN export BOOST_VERSION_UNERLINE=$(echo ${BOOST_VERSION} | sed 's/\./_/g') && \
...
-    make -j $(nproc) && \
-    make install && \
-    ldconfig && \
-    cd .. && \
-    rm -rf mpich-3.1.4*
-
 ENV CMAKE_PREFIX_PATH=/usr/local/lib/cmake
 
-RUN git clone -b release_v1.1 https://github.com/GridTools/gridtools.git && \
+RUN git clone -b release_v2.0 https://github.com/GridTools/gridtools.git && \
```
Comment (fthaler) [inline Dockerfile:31]: Could be fixed to 2.0.1 once released.

File: Dockerfile [hunk 3]
```diff
@@ -24,32 +26,24 @@ RUN export BOOST_VERSION_UNERLINE=$(echo ${BOOST_VERSION} | sed 's/\./_/g') && \
...
     mkdir -p gridtools/build && \
     cd gridtools/build && \
+    if [ -d /opt/rocm ]; then export CXX=/opt/rocm/bin/hipcc; fi && \
     cmake \
     -DCMAKE_BUILD_TYPE=Release \
-    -DBUILD_TESTING=OFF \
+    -DBUILD_TESTING=Off \
```
Comment (fthaler) [inline Dockerfile:37]:
```suggestion
    -DBUILD_TESTING=OFF \
```

File: Dockerfile [hunk 4]
```diff
@@ -47,9 +40,10 @@ RUN git clone -b release_v1.1 https://github.com/GridTools/gridtools.git && \
     cd ../.. && \
     rm -rf gridtools
 
-RUN git clone https://github.com/GridTools/GHEX.git && \
+RUN git clone -b gt2 https://github.com/fthaler/GHEX.git && \
```
Comment (fthaler) [inline Dockerfile:43]: TODO: point back to master (or release tag once available!) once GHEX was upgraded to GT2.

File: common/types.hpp [hunk 5]
```diff
@@ -22,24 +38,20 @@ constexpr real_t operator"" _r(unsigned long long value) {
 
 static constexpr gt::int_t halo = 3;
 
-#ifdef GTBENCH_BACKEND_CUDA
-template <gt::int_t IBlockSize = 32, gt::int_t JBlockSize = 8>
```
Comment (fthaler) [inline common/types.hpp:26]: Block-size tuning was removed here. Architecture-specific block sizes and/or block sizes as CMake variables will be reintroduced in a future PR.

**Review by boeschf — APPROVED** [review]

**Pull request merged**